com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/resource/AbstractResourceDescriptionDelta.java

```diff
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Avaloq Evolution AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Avaloq Evolution AG - initial API and implementation
+ *******************************************************************************/
+
+package com.avaloq.tools.ddk.xtext.resource;
+
+import java.util.Map;
+
+import org.eclipse.xtext.resource.IResourceDescription.Delta;
+
+import com.google.common.collect.Maps;
+
+
+/**
+ * Support for extensions on resource description delta.
+ */
+public abstract class AbstractResourceDescriptionDelta implements Delta {
+
+  private final Map<Class<? extends Object>, Object> extensions = Maps.newHashMap();
+
+  /**
+   * Adds the extension data.
+   *
+   * @param data
+   *          the data
+   */
+  public void addExtensionData(final Object data) {

```

Comment:
IMHO it would make sense to have an explicit "key" parameter. That would allow using different implementation classes and the client calling getExtensionData() doesn't need t know which the implementation class is.

Reply:
Sure. As Maxim pointed out I don't really need the map here. I want to attach a bit of payload in a way this class doesn't know the payload, but only the consumer (builder in this case). It simply allows to avoid API changes for builder participants as delta will carry the attached objects to the right place. Maybe some ideas how to avoid doing the generic extension thing?